meteor-app/imports/ui/pages/workspace/component.js

```diff
@@ -1,6 +1,65 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Line } from 'react-chartjs-2';
+import ifvisible from 'ifvisible.js';
+
+let theWindow = null;
+
+/**
+ * Helper function to minimize a window.
+ * @param {Window} w
+ */
+function Minimize (w) {
+  // There's no way of truely minimizing the window.
+  // The work-around here is to move it out of the screen.
+  w.blur();
+  w.resizeTo(0, 0);
+  w.moveTo(screen.width, screen.height);
+}
+
+/**
+ * Helper function to restore a minimized window.
+ * @param {Window} w
+ */
+function RestoreMinimized(w) {
+  const { width, height, x, y } = w._minimizeRestore;
+  w.moveTo(x, y);
+  w.resizeTo(width, height);
+  w.focus();
+}
+
+function openWindow(coord) {
+  if (theWindow) {
+    theWindow.close();
+  }
+
+  theWindow = window.open(`/workspace/charts?longitude=${coord[0]}&latitude=${coord[1]}`, '_blank', 'height=600,width=800,menubar=no,status=no,titlebar=no');
+
+  theWindow.onfocus = () => {
+    RestoreMinimized(theWindow);
+  };
+
+  theWindow._minimizeRestore = {
+    width: theWindow.outerWidth,
+    height: theWindow.outerHeight,
+    x: theWindow.screenX,
+    y: theWindow.screenY,
+  };
+}
+
+// Minimize the child window when the parent window becomes inactive
+ifvisible.on('blur', function () {
```

Comment:
These bindings should not happen globally. They should take effect **only** when this component is rendered, and everything should be properly cleaned up when this component exists.